src/Libraries/Microsoft.Extensions.AI.Abstractions/SpeechToText/SpeechToTextClientExtensions.cs

```diff
@@ -0,0 +1,81 @@
+    /// including itself or any services it might be wrapping.
+    /// </remarks>
+    public static TService? GetService<TService>(this ISpeechToTextClient client, object? serviceKey = null)
+    {
+        _ = Throw.IfNull(client);
+
+        return (TService?)client.GetService(typeof(TService), serviceKey);
+    }
+
+    /// <summary>Generates text from speech providing a single audio speech <see cref="DataContent"/>.</summary>
+    /// <param name="client">The client.</param>
+    /// <param name="audioSpeechContent">The single audio speech content.</param>
+    /// <param name="options">The speech to text options to configure the request.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>The text generated by the client.</returns>
+    public static async Task<SpeechToTextResponse> GetTextAsync(
+        this ISpeechToTextClient client,
+        DataContent audioSpeechContent,
+        SpeechToTextOptions? options = null,
+        CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(client);
+        _ = Throw.IfNull(audioSpeechContent);
+
+        using var audioSpeechStream = MemoryMarshal.TryGetArray(audioSpeechContent.Data, out var array) ?
+            new MemoryStream(array.Array!, array.Offset, array.Count) :
+            new MemoryStream(audioSpeechContent.Data.ToArray());
+
+        return await client.GetTextAsync(audioSpeechStream, options, cancellationToken).ConfigureAwait(false);
+    }
+
+    /// <summary>Generates text from speech providing a single audio speech <see cref="DataContent"/>.</summary>
+    /// <param name="client">The client.</param>
+    /// <param name="audioSpeechContent">The single audio speech content.</param>
+    /// <param name="options">The speech to text options to configure the request.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>The text generated by the client.</returns>
+    public static async IAsyncEnumerable<SpeechToTextResponseUpdate> GetStreamingTextAsync(
+        this ISpeechToTextClient client,
+        DataContent audioSpeechContent,
+        SpeechToTextOptions? options = null,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(client);
+        _ = Throw.IfNull(audioSpeechContent);
+
+        using var audioSpeechStream = MemoryMarshal.TryGetArray(audioSpeechContent.Data, out var array) ?
+            new MemoryStream(array.Array!, array.Offset, array.Count) :
+            new MemoryStream(audioSpeechContent.Data.ToArray());
+
+        await foreach (var update in client.GetStreamingTextAsync(audioSpeechStream, options, cancellationToken).ConfigureAwait(false))
```

Comment:
Same here. This can just return GetStreamingTextAsync, after removing async, removing EnumeratorCancellation, and removing the using.